Framework/Kernel/inc/MantidKernel/Cache.h

```diff
@@ -123,6 +167,7 @@ template <class KEYTYPE, class VALUETYPE> class DLLExport Cache {
     bool isValid = it_found != m_cacheMap.end();
 
     if (isValid) {
+      std::lock_guard<std::mutex> lock(m_mutex);
```

Comment:
What is this supposed to do? It does not seem to protect against modification of the map after the `find`?